rscripts/generate_stats_figs.R

```diff
@@ -1,6 +1,31 @@
+# load getopt to allow use of get_Rscript_filename for provenance-gathering
 suppressMessages(library(getopt, quietly=T))
-# Determine location of provenance.R relative to current file
-source(file.path(dirname(getopt::get_Rscript_filename()), "provenance.R"))
+
+# check result of get_Rscript_filename(); will be NA if running interactively
+this_script <- getopt::get_Rscript_filename()
+
+# Handle interactive session behaviours while getting Rscript filename:
+# if results of get_Rscript_filename() returns NA (ie interactive session)
+# then set variable this_script to "generate_stats_figs.R" to resolve breakage
+HandleRscriptFilename <- function() {
+  if (is.na(this_script)){ 
+    # if interactivity breaks get_Rscript_filename()
+    this_script <<- "generate_stats_figs.R"
+    return(this_script)
+  }
+}
+# run function to obtain name of this script (interactivity fix)
+HandleRscriptFilename()
+
+# Specify full path to rscripts folder
+path_to_rscripts <- file.path("~", "RiboViz","rscripts")
```

Comment:
This requires that the user has cloned "riboviz" into their home directory. This may not be the case and, if not, the user will get an error e.g.

```console
$ pwd
/home/ubuntu/develop/riboviz
$ Rscript --vanilla rscripts/generate_stats_figs.R --num-processes=1 --min-read-length=10 --max-read-length=50 --buffer=250 --primary-id=Name --dataset=vignette --hd-file=vignette/output/WTnone/WTnone.h5 --orf-fasta-file=vignette/input/yeast_YAL_CDS_w_250utrs.fa --rpf=True --output-dir=vignette/output/WTnone --do-pos-sp-nt-freq=True --t-rna-file=data/yeast_tRNAs.tsv --codon-positions-file=data/yeast_codon_pos_i200.RData --features-file=data/yeast_features.tsv --orf-gff-file=vignette/input/yeast_YAL_CDS_w_250utrs.gff3 --asite-disp-length-file=data/yeast_standard_asite_disp_length.txt --count-threshold=64
Error in file(filename, "r", encoding = encoding) : 
  cannot open the connection
Calls: source -> file
In addition: Warning message:
In file(filename, "r", encoding = encoding) :
  cannot open file '/home/ubuntu/RiboViz/rscripts/provenance.R': No such file or directory
Execution halted
```

From Google, R supports an `interactive()` function that returns TRUE or FALSE. A more robust solution might be something like:

```
if (interactive()) {
    # Use hard-coded script name and assume script is in "rscripts"
    # directory. This assumes that interactive R is being run within
    # the parent of rscripts/ but imposes no other constraints on
    # where rscripts/ or its parents are located.
    this_script <- "generate_stats_figs.R"
    path_to_this_script <- file.path("rscripts", this_script)
    source(file.path("rscripts", "provenance.R"))
} else {
    # Deduce file name and path using reflection as before.
    this_script <- getopt::get_Rscript_filename()
    path_to_this_script <- this_script
    source(file.path(dirname(this_script), "provenance.R"))
}
```

Reply:
This is a way better solution, you're right. I hadn't considered checking for a function like interactive(). Happy to include this fix instead. Good call re: home directory presumption too. 

Shall I change the branch and re-submit the pull request?